src/app/unlocked-stage/unlocked-stage.component.ts

```diff
@@ -1,156 +1,331 @@
+  private _humanBarWidth = 100;
+  private _mousemoveListener;
+  private _mouseleaveListener;
+  private _mousedownListener;
+  private _mouseupListener;
+  private _mousedown = false;
+  private _status: object = {stage: 5, activity: 3};
+  private _timeout;
+  private $mosquito;
+  private $spray;
+  private $net;
+  private mosquitoes = [];
+  private maxMosquitoes = 10;
+  private mosquitoesStrength = Array(this.maxMosquitoes).fill(100);
+  private mosCountInitial = 2;
+  private mosCountCurrent = this.mosCountInitial;
+  private keydown = false;
+  private levelCounter = 1;
+  private maxLevel = 4;
+
+  constructor(private _langService: LanguageService, private _renderer: Renderer2, private _dashboardService: DashboardService, private _sharedData: SharedDataService,
+              public snackBar: MatSnackBar) {
+  }
 
-    attack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray.gif';
-        this._mousedown = true;
+  // If we need keyboard movement for the net
+  @HostListener('window:keyup', ['$event'])
+  keyEvent(event: KeyboardEvent) {
+
+    if (event.keyCode === 38) {
+      this.keydown = true;
+      const that = this;
+      setInterval(function () {
+        if (that.keydown) {
+          that.$net.css({top: '-=2px'});
+        }
+      }, 10);
     }
+    if (event.keyCode === 40) {
+      this.keydown = false;
+      this.$net.css({top: '+=20px'});
+    }
+  }
+
+
+  ngOnInit() {
+    this._langService.loadLanguage().subscribe(response => {
+      this.language = response.pcprepkit.stages.lockedStage.game;
+    });
+    this._canvas = document.getElementById('scene');
+
+    // Initialising single mosquito
+    this.$mosquito = $('<div><img src="../assets/img/unlocked-stage/mosquito.gif" class="moz"></div>');
+    const progress = $('<div class="mosquitoHealth">' + 100 + '</div>');
+    this.$mosquito.attr({'class': 'moz'});
+    this.$mosquito.append(progress);
 
-    stopAttack() {
-        this._spray.src = '../assets/img/unlocked-stage/spray-cursor.gif';
-        this._mousedown = false;
+    // Adding n = mosCountInitial initially in the game
+    for (let i = 1; i <= this.mosCountInitial; i++) {
+      const mosquitoClone = this.$mosquito.clone();
+      $('#scene').append(mosquitoClone);
+      this.mosquitoes.push(mosquitoClone);
     }
 
-    moveOverCanvas(e) {
-        this._canvas.style.cursor = 'none';
-        this._spray.style.display = 'block';
-        this._spray.style.left = e.pageX - 425 + 'px';
-        this._spray.style.top = e.pageY - 150 + 'px';
+    // Spray element
+    this.$spray = $('<img src="../assets/img/unlocked-stage/spray-cursor.gif">');
+    this.$spray.attr({'id': 'spray'});
+    $('#scene').append(this.$spray);
+
+    // Net element
+    this.$net = $('<img src="../assets/img/unlocked-stage/net.png">');
+    this.$net.attr({'id': 'net'});
+    $('#scene').append(this.$net);
+    this.$net.css({display: 'block'});
+    this.$net.css({left: '840px'});
```

Comment:
% instead of px. It isn't responsive right now. Try zooming the screen to see the issue.

Reply:
Yup. I fixed the issue with the net.